lib/AppConfig.php

```diff
@@ -89,7 +89,19 @@ public function setUserValue($userId, $key, $value) {
 	 * @return bool
 	 */
 	public function enterpriseFeaturesEnabled() {
-		if (!$this->licenseManager->checkLicenseFor($this->appName)) {
+		$isEnterprise = $this->config->getAppValue($this->appName, 'isEEActive', null);
+		if ($isEnterprise === null) {
+			// app not marked as EE -> return false without doing anything more
+			return false;
+		}
+
+		$isEnterprise = \filter_var($isEnterprise, FILTER_VALIDATE_BOOLEAN);
+		if (!$isEnterprise || !$this->licenseManager->checkLicenseFor($this->appName)) {
+			// if not marked as enterprise, no need to check the license:
+			// explicitly mark the app as non-enterprise and remove the secure view.
+			// if marked as enterprise, we need to verify the license
+			$this->config->setAppValue($this->appName, 'isEEActive', 'no');
```

Comment:
lets align with true/false ?

Reply:
My main worry is with the type-checking. The expectation is that the admin runs the `occ config:app:set richdocuments isEEActive --value yes` command. I guess this will be evaluated and saved as string. Even if they set `true` or `false` as value, it will be a string. This might be confusing if we get a boolean as default because we'll need to deal with true/false as boolean and also with true/false as string. Note that `"false"` == `true`

By using yes/no, it's more clear that they'll always be string.
In addition, DB value is also more clear (it will show yes/no instead of 1 and empty string) and also is aligned with other similar keys.

Reply:
Yes you are right, we should change this for other setting too